### ISSUE TITLE
Dry run wedged with partially complete job

### ISSUE THREAD

**Issue by JohnDMcMaster** [SiliconAnalysis/xystitch#12]
TLDR: this may be an internal python bug based off of "SystemError: NULL result without error in PyObject_Call". But maybe I've done something else wrong, and this was just the symptom

Something happened after a large run failed and I tried to kick if off again by patching in smaller supertiles to fill in the failed very large supertile. I was able to simplify the test case to 2 workers with the following observation:
-Worker 1 as expected: got jobs, completed them, and returned them to master. Master processed completions ok
-Worker 0 got jobs, completed them, and returned them to master. However, master refused to process completions
-self.workers[0].qo.qsize(): consistently returned a large number (say 21)
-self.workers[0].qo.get(False) always threw Queue.Empty, even though there was clearly data
-self.workers[0].qo.get_nowait() gave the same result
-At one point I saw "SystemError: NULL result without error in PyObject_Call". This is really suspicious
-Restarting the system didn't solve the issue
-Was solved by clearing the existing supertile and moving to a smaller supertile size
-Other: the supertile was generated, but tiles were not. I didn't look into what the original failure was that might have caused that
-System has plenty of RAM and disk space (100+GB of each free)
